### PR TITLE
test(workspaces): add workspace matrix playwright e2e tests

### LINKED ISSUE #2443: Adapt workspace e2e tests for OpenShell compatibility
### Task content

Review and update the existing workspace e2e test suite to ensure it aligns with OpenShell specifications and runs successfully.

### LINKED ISSUE #2444: Expand workspace e2e tests: filesystem & networking combinations
### Task content

Extend the existing workspace test suite to cover various combinations of filesystem types and networking configurations during workspace creation.

### PR DESCRIPTION
**Pull request by serbangeorge-m**
## Summary
- Add filesystem × network sandbox matrix E2E tests for OpenCode and Claude Code agents. The matrix covers 4 file-access levels (none, home, custom paths, full-system) × 3 network modes (deny-all, developer preset, unrestricted) plus edge cases (read-only mounts, multi-mounts, custom hosts) — 16 scenarios × 2 agents × 5 lifecycle steps each, with known-broken combos (full-system, unrestricted) auto-skipped
- Extract shared lifecycle helper and sandbox matrix registration into `helpers/`, with UI-aligned tags (`@fs-*`, `@net-*`) for granular `--grep` filtering
- Add compact terminal reporter, workspace page objects (create wizard, terminal, overview), and workspace provider E2E documentation

Closes https://github.com/openkaiden/kaiden/issues/2443
Closes https://github.com/openkaiden/kaiden/issues/2444

## Test plan
- [x] Run `pnpm exec playwright test --project=Workspace-Provider --list` — confirms tests are discovered with `PODMAN_ENABLED=true` and `WORKSPACE_TESTS_CI=true`
- [x] Run minimal smoke: `--grep "FS-NONE-NET-DEVELOPER"` — passes for both agents
- [x] Run full matrix: `--project=Workspace-Provider --timeout=600000` — most pass, full-system and unrestricted scenarios skip as expected
- [x] Verify existing agent lifecycle specs (`workspace-opencode-smoke`, `workspace-claude-smoke`, etc.) still pass
- [x] Verify `Kaiden-App-Core` project tests unaffected

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/playwright/src/model/pages/agent-workspace-terminal-page.ts [hunk 1]
```diff
@@ -27,6 +27,11 @@ export class AgentWorkspaceTerminalPage extends BasePage {
   readonly terminalContainer: Locator;
   readonly emptyTerminalMessage: Locator;
 
+  private static readonly ANTHROPIC_API_KEY_PROMPT = /Do you want to use this API key\?/i;
```
Comment (jeffmaury) [inline tests/playwright/src/model/pages/agent-workspace-terminal-page.ts:30]: **question:** do we need this ? I thought it was removed ?

Comment (serbangeorge-m) [reply tests/playwright/src/model/pages/agent-workspace-terminal-page.ts:30]: if it was removed, no we do not need it, will check

Comment (serbangeorge-m) [reply tests/playwright/src/model/pages/agent-workspace-terminal-page.ts:30]: apparently was removed only for vertex but not for claude, so we still need it

**Review by jeffmaury — APPROVED** [review]
